firmware/arduino/chessboard/circleFunction.ino

```diff
@@ -0,0 +1,250 @@
+            slopeIndex = SLOPE_HORIZONTAL;
+        else
+            slopeIndex = NUM_SLOPES_PER_QUARTER_CIRCLE - 1;
+
+        // Iterate through pulsesPerSlope
+        while (slopeIndex >= SLOPE_HORIZONTAL_MAX_SIZE  &&  slopeIndex <= SLOPE_VERTICAL)
+        {
+            // Y-scale
+            if (slopeIndex == SLOPE_TWO || slopeIndex == SLOPE_QUARTER_MAX_SIZE)
+                setScale(yMotor, QUARTER_STEPS);    
+            else if (slopeIndex == SLOPE_FOUR || slopeIndex == SLOPE_HALF_MAX_SIZE)
+                setScale(yMotor, HALF_STEPS);    
+            else if (slopeIndex == SLOPE_EIGHT || slopeIndex == SLOPE_VERTICAL_MAX_SIZE || slopeIndex == SLOPE_ONE_MAX_SIZE ||
+                     slopeIndex == SLOPE_FOUR_MAX_SIZE || slopeIndex == SLOPE_TWO_MAX_SIZE)
+                setScale(yMotor, WHOLE_STEPS);    
+            else
+                setScale(yMotor, EIGHTH_STEPS);    
+
+            // X-scale
+            if (slopeIndex == SLOPE_EIGHTH || slopeIndex == SLOPE_HORIZONTAL_MAX_SIZE || slopeIndex == SLOPE_ONE_MAX_SIZE ||
+                slopeIndex == SLOPE_QUARTER_MAX_SIZE || slopeIndex == SLOPE_HALF_MAX_SIZE)
+                setScale(xMotor, WHOLE_STEPS);
+            else if (slopeIndex == SLOPE_QUARTER || slopeIndex == SLOPE_TWO_MAX_SIZE)
+                setScale(xMotor, HALF_STEPS);
+            else if (slopeIndex == SLOPE_HALF || slopeIndex == SLOPE_FOUR_MAX_SIZE)
+                setScale(xMotor, QUARTER_STEPS);
+            else 
+                setScale(xMotor, EIGHTH_STEPS);
+
+            // Send pulses for the current slopeIndex
+            // Separate loops for vertical and horizontal cases to optimize performance
+            if (slopeIndex == SLOPE_VERTICAL || slopeIndex == SLOPE_VERTICAL_MAX_SIZE)
+            {
+                for (int i = 0; i < pulsesPerSlope[circle][slopeIndex]; i++)
+                {
+                    // Y-Step
+                    digitalWrite(yMotor[STEP_PIN], LOW);
+                    digitalWrite(yMotor[STEP_PIN], HIGH);
+                }
+            }
+            else if (slopeIndex == SLOPE_HORIZONTAL || slopeIndex == SLOPE_HORIZONTAL_MAX_SIZE)
+            {
+                for (int i = 0; i < pulsesPerSlope[circle][slopeIndex]; i++)
+                {
+                    // X-Step
+                    digitalWrite(xMotor[STEP_PIN], LOW);
+                    digitalWrite(xMotor[STEP_PIN], HIGH);
+                }
+            }
+            else
+            {
+                for (int i = 0; i < pulsesPerSlope[circle][slopeIndex]; i++)
+                {
+                    // X-Step
+                    digitalWrite(xMotor[STEP_PIN], LOW);
+                    digitalWrite(xMotor[STEP_PIN], HIGH);
+
+                    // Y-Step
+                    digitalWrite(yMotor[STEP_PIN], LOW);
+                    digitalWrite(yMotor[STEP_PIN], HIGH);
+                }
+            }
+
+            // Increment or Decrement slopeIndex
```

Comment:
This comment should say something about why you are doing it rather than what:
```
// If in quadrant 2 or 4, slope is increasing, otherwise it's increasing.
```